api/server.go

```diff
@@ -0,0 +1,43 @@
+package server
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+
+	"github.com/darkjedidj/cinema-service/api/halls"
+	"github.com/gorilla/mux"
+	"github.com/joho/godotenv"
+)
+
+type App struct {
+	Router *mux.Router
+}
+
+// New creates router with handler
+func (a *App) New(db *sql.DB) {
+
+	myRouter := mux.NewRouter().StrictSlash(false)
+	myRouter.HandleFunc("/v1/halls/{id}", halls.Handle(db))
```

Comment:
```suggestion
	myRouter.HandleFunc("/v1/halls/{id}", &halls.Handler{&hall.Repo{db}}.Handle())
```

I know that this is not working, but move to this direction

Reply:
like, i pushed my vision of solving it, but the error doesnt match problem in the code i guess 
![image](https://user-images.githubusercontent.com/70835145/151378221-220a200a-bec2-4504-b723-c27cba1ae013.png)

Reply:
yeah, and i cant understand why we are using "&" before first package, why we are taking adress, we appeal to the method, not to the pointer

Reply:
The solution is not about syntax, what I was trying to say is having a struct for the handler, where u have a struct for the Repo, which contains the db